### PR TITLE
Feature/player level indicator

### LINKED ISSUE #48: Player level indicator
The player's item on the board should have indication of their level:
- visibly via the avatar/icon colour matching the level colour
- perhaps a tool tip or some other indicator for the color-blind users

### PR DESCRIPTION
**Pull request by UserC2**
Fixes #48
* Player avatar icon colour changes according to level
* A tooltip over the icon shows whether the player is a member or guest and displays their level
* Mocked players are assigned a level

### CODE REVIEW

File: vueapp/src/models/player.ts [hunk 1]
```diff
@@ -15,7 +16,9 @@ export class Player {
   }
 
   static generateMockPlayers(n: number, label: string = "Mock Player", start: number = 1): Player[] {
-    return Array.from({ length: n }, (v, i) => i).map(i => new Player(`${label} ${i + start}`))
+    var levelStore = useLevelStore();
+    levelStore.loadLevels();
```
Comment (UserC2) [inline vueapp/src/models/player.ts:20]: This doesn't seem to be the correct way to use the level store. Is it fine to do this every time it's needed or should it be loaded in some central location then accessed through that?

File: vueapp/src/components/Player.vue [hunk 2]
```diff
@@ -3,6 +3,8 @@
   <template v-slot:prepend>
     <v-icon
       :icon="player.isGuest ? 'mdi-account-box-outline' : 'mdi-account-circle'"
+      :color="player?.level?.color"
+      v-tooltip="(player.isGuest ? 'Member (' : 'Guest (') + (player?.level?.name || 'No Level') + ')'"
```
Comment (l2lam) [inline vueapp/src/components/Player.vue:7]: String concatenation are bloody hard to read and hell to translate.  Use string formatting and whole phrases for each case.

File: vueapp/src/models/player.ts (hunk 1 above)
Comment (l2lam) [reply vueapp/src/models/player.ts:20]: Perhaps we can be a little smarter about when to load the levels (an other data)... like if the store detects if the levels have been loaded (perhaps if there are any levels) and load it if necessary.  So `allLevels` become a calculated property and does the checking for us.

File: vueapp/src/components/Player.vue (hunk 2 above)
Comment (l2lam) [reply vueapp/src/components/Player.vue:7]: And the logic is reversed
```suggestion
      v-tooltip="(!player.isGuest ? 'Member (' : 'Guest (') + (player?.level?.name || 'No Level') + ')'"
```

File: vueapp/src/components/Player.vue [hunk 3]
```diff
@@ -3,6 +3,8 @@
   <template v-slot:prepend>
     <v-icon
       :icon="player.isGuest ? 'mdi-account-box-outline' : 'mdi-account-circle'"
+      :color="player?.level?.color"
```
Comment (l2lam) [inline vueapp/src/components/Player.vue:6]: When we edit a Guest and change their level, the icon displays as 'no level' even though the the editor shows the correct level.

And in the member's editor the color indicator is not shown.

File: vueapp/src/components/Player.vue (hunk 2 above)
Comment (UserC2) [reply vueapp/src/components/Player.vue]: Fixed in 7890747f7080ac53682028131bcbd5513be0b3ec

File: vueapp/src/components/Player.vue (hunk 3 above)
Comment (UserC2) [reply vueapp/src/components/Player.vue]: Within `GuestUpsert.vue` and `MemberUpsert.vue` the player's level isn't being set correctly (maybe something to do with how `v-model` is setting it?).
Printing the player's level to the console shows that it is only updated in `v-select`'s model but not in the store (where it is set to undefined).

Comment (l2lam) [reply vueapp/src/components/Player.vue]: The `GuestUpsert` selection control is fixed in the latest commit (missing the `return-object` attribute).  The `MemberUpsert` doesn't support changing levels at this time.

**Pull request merged**